shell/scl_source

```diff
@@ -7,7 +7,7 @@ Don't use this script outside of SCL scriptlets!
 Options:
     -h, --help    display this help and exit"
 
-if [ $# -eq 0 -o $1 = "-h" -o $1 = "--help" ]; then
+if [ $# -eq 0 -o X"$1" = X"-h" -o X"$1" = X"--help" ]; then
```

Comment:
This is not a bad fix in general, thanks!  But could you change the return status to non-zero value in case of  `test $# -eq 0` please?